MBS-11505 keyboard checks rework

RuslanMingaliev opened the pull request.

RuslanMingaliev committed:
MBS-11505 Rework keyboard checks

RuslanMingaliev committed:
MBS-11505 Merge branch 'develop' into MBS-11505-keyboard-checks-rework

RuslanMingaliev commented on subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/KeyboardTest.kt:
```diff
@@ -0,0 +1,230 @@
+    }
+
+    @Test
+    fun openActivity_keyboardIsNotDisplayed_whenAdjustUnspecified() {
```
IDK how to make these tests better without Parameterized mechanism

sergioboi replied:
The proposition is to make a few classes and split cases i.e. by Adjust type
And make a Base TestClass wherein constructor we will pass a type parameter

RuslanMingaliev replied:
Nice idea!

eugene-krivobokov replied:
Does it work? See MBS-10589

RuslanMingaliev replied:
Unfortunately, no. It doesn't work

sergioboi replied:
What about place test methods in the base class and override them to hack runner

RuslanMingaliev commented on subprojects/android-test/ui-testing-core-app/src/main/kotlin/com/avito/android/ui/KeyboardActivity.kt:
```diff
@@ -0,0 +1,33 @@
+        input.clearFocus()
+    }
+
+    fun openKeyboard() {
```
These functions are mostly copied from the Avito app source code

RuslanMingaliev commented on subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/page_object/KeyboardElement.kt:
```diff
@@ -26,98 +27,65 @@ class KeyboardElement : PageObject() {
+         */
+        fun isDisplayed(@Suppress("UNUSED_PARAMETER") activity: Activity?) = isDisplayed()
 
-            InstrumentationRegistry.getInstrumentation().runOnMainSync {
```
I know that this is a more honest way, but I don't know how to do it more stable. The new one seems to be working.

eugene-krivobokov replied:
Did you evaluate flakiness of the new implementation?
E.g. run the same test dozens of times in a row.
If I'm not mistaken, `UiDevice` invocations were not so stable.

RuslanMingaliev commented on subprojects/android-test/ui-testing-core-app/src/androidTest/kotlin/com/avito/android/ui/test/Screen.kt:
```diff
@@ -34,6 +34,9 @@ object Screen {
     val identicalCellsRecycler: IdenticalCellsRecyclerScreen
         get() = IdenticalCellsRecyclerScreen()
 
+    val keyboardScreen: KeyboardScreen
```
A new instance created on every access to this property. That really what we want? Or it can be wrapped into `lazy` for example?

eugene-krivobokov replied:
Does it heavy or has a state?

sergioboi commented on subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/page_object/KeyboardElement.kt:
```diff
@@ -26,98 +27,65 @@ class KeyboardElement : PageObject() {
+                output = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+                    .executeShellCommand(KEYBOARD_STATE_CHECK_CMD)
+            } catch (e: IOException) {
+                throw RuntimeException("Keyboard check failed", e)
```
Fail while executing KEYBOARD_STATE_CHECK_CMD

sergioboi commented on subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/page_object/KeyboardElement.kt:
```diff
@@ -26,98 +27,65 @@ class KeyboardElement : PageObject() {
 
-                threshold = (activityHeight - minimalKeyboardHeight).toInt()
+        fun isNotDisplayed() {
+            waitForAssertion {
```
What is a default timeout for waiting?

eugene-krivobokov commented on subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/page_object/KeyboardElement.kt:
```diff
@@ -26,98 +27,65 @@ class KeyboardElement : PageObject() {
-            // not a bad name for private const
-            @Suppress("VariableMaxLength")
-            private const val KEYBOARD_MINIMUM_HEIGHT_PERCENTAGE = .30f
+            private const val KEYBOARD_STATE_CHECK_CMD = "dumpsys input_method | grep mInputShown"
```
Is output the same on all API versions?

RuslanMingaliev committed:
MBS-11505 Fix code review issues

dsvoronin committed:
MBS-11504 bump release version (#1094)

dsvoronin committed:
MBS-10193 move gradleTestRuntime workaround to convention plugin (#1095)

dsvoronin committed:
MBS-7002 promote clear-package failed attempt to warning in logger, increase timeout (#1097)

raw command line output visible through exception message

Reason: there is a huge error rate on this command, needs investigation

Retries helps (only 1 real failure in 24h)

dsvoronin committed:
MBS-11493 add `make fast_check` to pre-comit hook; HowToStart docs (#1093)

mention makefile

dsvoronin committed:
MBS-11456 update to AGP 4.2, support signer plugin (#1091)

dsvoronin committed:
MBS-9699 explicitApi in random-utils (#1099)

dsvoronin committed:
MBS-9085 enable configuration cache (#1098)

add docs for opt-out
enable cache compatibility tests

dsvoronin committed:
MBS-9085 configuration cache opt-in (#1100)

dsvoronin committed:
MBS-10193 bump release version (#1101)

RuslanMingaliev committed:
MBS-11505 Merge branch 'develop' into MBS-11505-keyboard-checks-rework

sergioboi commented:
@RuslanMingaliev propose you to delete branch and close this PR
